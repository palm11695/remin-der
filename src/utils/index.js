export function DateTimeFormatter(props) {
  const { date } = props
  const dateObj = new Date(date)

  return (
    <div className="pb-4">
      <span className="font-bold">Due Date: </span>
      <span>{`${
        LeadingZero({number: dateObj.getDate()}) +
        "/" +
        LeadingZero({number: dateObj.getMonth()}) +
        "/" +
        dateObj.getFullYear()
        } - ${
          LeadingZero({number:dateObj.getHours()}) +
          ":" +
          LeadingZero({number: dateObj.getMinutes()})
        }`}
      </span>
    </div>
  );
}

export function LeadingZero({number}) {
  return parseInt(number) < 10 ? "0" + number : number
}