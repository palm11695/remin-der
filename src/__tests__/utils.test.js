import { create } from 'react-test-renderer';
import { DateTimeFormatter, LeadingZero } from '../utils';

/*
* @Test DateTimeFormatter
* [Apply ISP]
* To check that it will return the correct date format
*/
describe('DateTimeFormatter', () => {
  /* 
  * Try to pass the current date to 'DateTimeFormatter'
  * Expected Result: It will return the correct date format with current date in snapshot.
  */
  it('should match when date is now', () => {
    const component = create(DateTimeFormatter({ date: new Date() }));
    expect(component.toJSON()).toMatchSnapshot();
  })

  /* 
  * Try to pass date: null to 'DateTimeFormatter'
  * Expected Result: It will return the correct date format (01/00/1970 - 07:00) in snapshot.
  */
  it('should match when date is null', () => {
    const component = create(DateTimeFormatter({ date: null }));
    expect(component.toJSON()).toMatchSnapshot();
  })
})

/*
* @Test LeadingZero
* [Apply ISP]
* To check that it will return leading zero when number value is less than 10 (To make the date format correct)
*/
describe('LeadingZero', () => {
  /* 
  * Try to pass number = 1 to 'LeadingZero'
  * Expected Result: It will return "01" as String in snapshot.
  */
  it('should match when number < 10', () => {
    const component = create(LeadingZero({ number: 1 }));
    expect(component.toJSON()).toMatchSnapshot();
  })

  /* 
  * Try to pass number = 10 to 'LeadingZero'
  * Expected Result: It will return "10" as String in snapshot.
  */
  it('should match when number >= 10', () => {
    const component = create(LeadingZero({ number: 10 }));
    expect(component.toJSON()).toMatchSnapshot();
  })
})
