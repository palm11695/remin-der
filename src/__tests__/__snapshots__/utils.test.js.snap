// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeFormatter should match the snapshot 1`] = `
<div
  className="pb-4"
>
  <span
    className="font-bold"
  >
    Due Date: 
  </span>
  <span>
    24/10/2022 - 00:04
  </span>
</div>
`;

exports[`DateTimeFormatter should match when date is null 1`] = `
<div
  className="pb-4"
>
  <span
    className="font-bold"
  >
    Due Date: 
  </span>
  <span>
    01/00/1970 - 07:00
  </span>
</div>
`;

exports[`LeadingZero should match when number < 10 1`] = `"01"`;

exports[`LeadingZero should match when number >= 10 1`] = `"10"`;
